### PR TITLE
pkp/pkp-lib#700 Clean up plugin API

### PR DESCRIPTION
**Pull request by asmecher**

### CODE REVIEW

File: CustomBlockManagerPlugin.inc.php [hunk 1]
```diff
@@ -119,53 +119,46 @@ function setupGridHandler($hookName, $params) {
...
-			import('lib.pkp.classes.linkAction.request.AjaxLegacyPluginModal');
-			$actionRequest = new AjaxLegacyPluginModal(
-					$router->url($request, null, null, 'plugin', null, array('verb' => 'manageCustomBlocks', 'plugin' => $this->getName(), 'category' => 'generic')),
-					$this->getDisplayName()
-			);
-			return new LinkAction($verbName, $actionRequest, $verbLocalized, null);
-		}
-
-		return null;
+		return array_merge(
```
Comment (beghelli) [inline CustomBlockManagerPlugin.inc.php]: will always add the manage action even if the plugin is disabled right?

File: CustomBlockManagerPlugin.inc.php [hunk 2]
```diff
@@ -119,53 +119,46 @@ function setupGridHandler($hookName, $params) {
...
-				$pluginModalContent = $form->fetch($request);
-				return true;
-			default:
-				return parent::manage($verb, $args, $message, $messageParams);
-		}
+	function manage($args, $request) {
+		$templateMgr = TemplateManager::getManager($request);
+		$dispatcher = $request->getDispatcher();
+		return $templateMgr->fetchAjax(
+			'customBlockGridUrlGridContainer',
```
Comment (beghelli) [inline CustomBlockManagerPlugin.inc.php]: is it ok coupling the template and the class like this?

Comment (beghelli) [inline CustomBlockManagerPlugin.inc.php]: oh, ok, it's not anywhere else. You just avoided creating a template just for calling the grid. OK. :)

Comment (asmecher) [inline CustomBlockManagerPlugin.inc.php]: Yup, I added fetchAjax to simplify that case. There were about a dozen templates like that.

**Pull request merged**